extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js

```diff
@@ -485,45 +548,58 @@ export class GoogleSubscriptionsPlatform {
     const encryptedDocumentKey = this.serviceAdapter_.getEncryptedDocumentKey(
       'google.com'
     );
-    return this.runtime_
-      .getEntitlements(encryptedDocumentKey)
-      .then((swgEntitlements) => {
-        // Get and store the isReadyToPay signal which is independent of
-        // any entitlments existing.
-        if (swgEntitlements.isReadyToPay) {
-          this.isReadyToPay_ = true;
-        }
+    userAssert(
+      !(this.enableLAA_ && encryptedDocumentKey),
+      `enableLAA cannot be used when the document is encrypted`
+    );
 
-        // Get the specifc entitlement we're looking for
-        let swgEntitlement = swgEntitlements.getEntitlementForThis();
-        let granted = false;
-        if (swgEntitlement && swgEntitlement.source) {
-          granted = true;
-        } else if (
-          swgEntitlements.entitlements.length &&
-          swgEntitlements.entitlements[0].products.length
-        ) {
-          // We didn't find a grant so see if there is a non granting
-          // and return that. Note if we start returning multiple non
-          // granting we'll need to refactor to handle returning an
-          // array of Entitlement objects.
-          // #TODO(jpettitt) - refactor to handle multi entitlement case
-          swgEntitlement = swgEntitlements.entitlements[0];
-        } else {
-          return null;
-        }
-        swgEntitlements.ack();
-        return new Entitlement({
-          source: swgEntitlement.source,
-          raw: swgEntitlements.raw,
-          service: PLATFORM_ID,
-          granted,
-          // if it's granted it must be a subscriber
-          grantReason: granted ? GrantReason.SUBSCRIBER : null,
-          dataObject: swgEntitlement.json(),
-          decryptedDocumentKey: swgEntitlements.decryptedDocumentKey,
+    return this.maybeGetLAAEntitlement_().then((laaEntitlement) => {
+      if (laaEntitlement) {
+        return laaEntitlement;
```

Comment:
Once the `getEntitlements` method accepts metering params, will this AMP code be updated to pass those params instead of immediately returning the LAA entitlement?

Reply:
Nevermind, I understand the requirements better now